seastar/src/timer.rs

```diff
@@ -106,3 +107,160 @@ fn dropper<Func>(raw_func: *mut u8) {
+                #[doc = concat!(" [`", stringify!($Timer), "::rearm`].")]
+                ///
+                /// # Arguments
+                /// * `delta` - The time when the timer expires, relative to now.
+                pub fn arm(&mut self, delta: $Duration) {
+                    self.arm_at(<$Clock>::now() + delta);
+                }
+
+                /// Sets the timer expiration time relatively to now. The timer will rearm
+                /// automatically with a period equal to `delta`.
+                ///
+                /// It is illegal to arm a timer that has already been armed (and not disarmed by
+                /// expiration or
+                #[doc = concat!(" [`", stringify!($Timer), "::cancel`]).")]
+                /// In the current implementation, this will result in an assertion failure. See
+                #[doc = concat!(" [`", stringify!($Timer), "::rearm_periodic`].")]
+                ///
+                /// # Arguments
+                /// * `delta` - The time when the timer expires, relative to now.
+                pub fn arm_periodic(&mut self, delta: $Duration) {
+                    self.arm_at_periodic(<$Clock>::now() + delta, delta);
+                }
+
+                /// Sets the timer expiration time relatively to now. If the timer was already then,
+                /// it is canceled first.
+                ///
+                /// # Arguments
+                /// * `delta` - The time when the timer expires, relative to now.
+                pub fn rearm(&mut self, delta: $Duration) {
+                    self.rearm_at(<$Clock>::now() + delta);
+                }
+
+                /// Sets the timer expiration time relatively to now. If the timer was already armed,
+                ///  it is canceled first. The timer will rearm automatically with a period equal
+                /// to `delta`.
+                ///
+                /// # Arguments
+                /// * `delta` - The time when the timer expires, relative to now.
+                pub fn rearm_periodic(&mut self, delta: $Duration) {
+                    self.rearm_at_periodic(<$Clock>::now() + delta, delta);
+                }
+
+                /// Returns whether the timer is armed.
+                pub fn armed(&self) -> bool {
+                    [<$ffi_pref _armed>](&self.inner)
+                }
+
+                /// Cancels an armed timer.
+                ///
+                /// If the timer was armed, it is disarmed. If the timer was not armed, does nothing.
+                ///
+                /// Returns `true` if the timer was armed before the call.
+                pub fn cancel(&mut self) -> bool {
+                    [<$ffi_pref _cancel>](self.inner.pin_mut())
+                }
+
+                /// Gets the expiration time of an armed timer.
+                pub fn get_timeout(&self) -> $Instant {
+                    $Instant { nanos: [<$ffi_pref _get_timeout>](&self.inner) }
+                }
```

Comment:
From the original docstring:
```rust
    /// \return the time at which the timer is scheduled to expire (undefined if the
    ///       timer is not armed).
```
While it's not undefined behavior, returning an undefined value is not a good API IMO. Let's slighly improve the semantics of the original function and return an Option here - if the timer is not armed then it will return None.